Shared/Actions/CreateMenuItem.swift

```diff
@@ -0,0 +1,507 @@
+import Foundation
+import AppIntents
+import SwiftUI
+
+// MARK: Action
+struct CreateMenuItem: AppIntent {
+	static let title: LocalizedStringResource = "Create Menu Item"
+
+	static let description = IntentDescription(
+"""
+Create a menu item with a title, subtitle and icon.
+
+You can later use one or more of these Items in a "Choose from List" action.
+
+Add an "Add to Variable" action below this to populate a list and then use that variable in the "Choose From List" action.
+""",
+		categoryName: "Rich Menu",
+		searchKeywords: [
+			"menu",
+			"menu item",
+			"choose from menu",
+			"rich menu"
+		]
+	)
+
+	static var parameterSummary: some ParameterSummary {
+		When(\.$menuTitle, .hasAnyValue) {
+			Switch(\.$iconType) {
+				// MARK: SFSymbol
+				Case(RMIconType.sfSymbol) {
+					When(\.$backgroundShape, .notEqualTo, .noBackground) {
+						Summary("Create \(\.$systemName) with \(\.$menuTitle) and \(\.$subtitle)") {
+							\.$iconType
+							\.$foreground
+							\.$backgroundShape
+							\.$background
+							\.$data
+						}
+					} otherwise: {
+						Summary("Create \(\.$systemName) with \(\.$menuTitle) and \(\.$subtitle)") {
+							\.$iconType
+							\.$foreground
+							\.$backgroundShape
+							\.$background
+							\.$data
+						}
+					}
+				}
+
+				// MARK: Emoji
+				Case(RMIconType.emoji) {
+					When(\.$backgroundShape, .notEqualTo, .noBackground) {
```

Comment:
Reverse the condition. Double-negative makes it harder to read.
```suggestion
					When(\.$backgroundShape, .equalTo, .noBackground) {
```